api-gateway/src/main/java/com/petclinic/bffapigateway/presentationlayer/v2/InventoryController.java

```diff
@@ -254,4 +254,29 @@ public Mono<ResponseEntity<byte[]>> createSupplyPdf(@PathVariable String invento
 
 
     }
+
+    @SecuredEndpoint(allowedRoles = {Roles.ADMIN, Roles.INVENTORY_MANAGER})
+    @PutMapping("/{currentInventoryId}/products/{productId}/updateInventoryId/{newInventoryId}")
+    @Operation(summary = "Update product's inventory ID in the inventory system")
+    @ApiResponses(value = {
+            @ApiResponse(description = "Inventory ID updated successfully", responseCode = "200"),
+            @ApiResponse(description = "Product not found", responseCode = "404"),
+            @ApiResponse(description = "Invalid inputs", responseCode = "400")
```

Comment:
Well done on the handling of the possible issues that might come up during the transition. Your usage of the @ApiResponse is really useful to return a complete solution that is handled by the system.